src/credentials/attestation_verifier/DeviceAttestationVerifier.cpp

```diff
@@ -87,6 +87,104 @@ DeviceAttestationVerifier * gDacVerifier = &gDefaultDACVerifier;
+    case AttestationVerificationResult::kFirmwareInformationMismatch:
+        return "Firmware information mismatch";
+    case AttestationVerificationResult::kFirmwareInformationMissing:
+        return "Firmware information missing";
+    case AttestationVerificationResult::kAttestationSignatureInvalid:
+        return "Attestation signature failed to validate against DAC subject public key";
+    case AttestationVerificationResult::kAttestationElementsMalformed:
+        return "Attestation elements payload is malformed";
+    case AttestationVerificationResult::kAttestationNonceMismatch:
+        return "Attestation nonce mismatches one from Attestation Request";
+    case AttestationVerificationResult::kAttestationSignatureInvalidFormat:
+        return "Attestation signature format is invalid (likely wrong signature algorithm in certificate)";
+    case AttestationVerificationResult::kCertificationDeclarationNoKeyId:
+        return "Certification declaration missing the required key ID in CMS envelope";
+    case AttestationVerificationResult::kCertificationDeclarationNoCertificateFound:
+        return "Could not find matching trusted verification certificate for the certification declaration's key ID";
+    case AttestationVerificationResult::kCertificationDeclarationInvalidSignature:
+        return "Certification declaration signature failed to validate against the verification certificate";
+    case AttestationVerificationResult::kCertificationDeclarationInvalidFormat:
+        return "Certification declaration format is invalid";
+    case AttestationVerificationResult::kCertificationDeclarationInvalidVendorId:
+        return "Certification declaration vendor ID failed to cross-reference with DAC and/or PAI and/or Basic Information cluster";
+    case AttestationVerificationResult::kCertificationDeclarationInvalidProductId:
+        return "Certification declaration product ID failed to cross-reference with DAC and/or PAI and/or Basic Information "
+               "cluster";
+    case AttestationVerificationResult::kCertificationDeclarationInvalidPAA:
+        return "Certification declaration required a fixed allowed PAA which mismatches the final PAA found";
+    case AttestationVerificationResult::kNoMemory:
+        return "Failed to allocate memory to process attestation verification";
+    case AttestationVerificationResult::kInvalidArgument:
+        return "Some unexpected invalid argument was provided internally to the device attestation procedure (likely malformed "
+               "input data from candidate device)";
+    case AttestationVerificationResult::kInternalError:
+        return "An internal error error arose in the device attestation procedure (likely malformed input data from candidate "
+               "device)";
+    case AttestationVerificationResult::kNotImplemented:
+        return "Reached a critical-but-unimplemented part of the device attestation procedure!";
+    default:
```

Comment:
Would be better to not have a default case here, so if a new result value gets added this code will fail to compile with our `-Werror` settings and need to be updated...

We can still have the fall-through to the code below for runtime-provided invalid values.